docs/src/decomposing-an-op-in-ttir.md

```diff
@@ -0,0 +1,53 @@
+# Decomposing an Op in TTIR
+
+This guide explains how to add and decompose a new operation end-to-end in tt-mlir. We’ll focus on adding an `Index` operation, which will be decomposed into the `Slice` operation. The decomposition is implemented as a conversion pass in MLIR since it allows us to mark operations or dialects as legal or illegal, type conversion...
```

Comment:
> This guide explains how to add and decompose a new operation end-to-end

I understood that this guide was going to explain how adding an op end-to-end should be done in tt-mlir. Maybe a little tweaking of the sentence would make it more accurate, something like `This guide explains how to add and decompose a new op in TTIR dialect.`.

Reply:
Yeah, I was not really precise in this explanation. I will refactor it, thanks for the heads up.